### ISSUE TITLE
[Bug]: Accessibility - Unable to select any item from the Dropdown with searchBox when Screen reader is present. 

### ISSUE THREAD

**Issue by NadRose** [microsoft/fluentui#22616]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
Test Environment:
Platform: Web- Edge
Browser: Edge Version 100.0.1185.50 (Official build) (64-bit)
Screen Reader: Narrator
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/nadrose/pen/ExozWZj?editors=1111

## Actual Behavior
When screen reader (Narrator) is present, and the dropdown contains searchBox, initial state - no item is selected. Navigate with keyboard and select Item from list with the Enter key (or Space key) does nothing.

## Expected Behavior
Navigate with keyboard and select Item from list should result in picking the item and set it to be the selected from dropdown list.

This behavior happen only when there are no items selected yet, screen reader is presented and only with Microsoft Edge browser.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
Thanks for filing this issue with us. To set expectations the developers will review this issue once capacity allows.

cc @smhigley - here is a reported a11y issue using Dropdown.

**Comment by smhigley** [comment]
The Dropdown does not support having a nested text input within it. This is a limitation of combobox semantics in general, and not just in Fluent. Doing so will cause unexpected and inaccessible results for assistive tech users.

In this particular case, I can get it to work by manually switching Narrator to application mode. That said, this is a power user interaction, and not something you should rely on -- which is to say, this is still a broken experience. The root cause is Narrator's built-in mode switching behavior based on the roles of the controls it is interacting with. The only fix is to not put a text input inside of the dropdown, since doing so creates invalid and unexpected semantics. I'd suggest using Combobox instead.

https://user-images.githubusercontent.com/3819570/168699022-616815b6-f97f-4123-a596-307dcaff52a7.mp4
